src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -143,7 +166,7 @@ private ReturnTypeChain BuildReturnTypes()
 
         private ConvenienceMethod BuildConvenienceMethod(ReturnTypeChain returnTypeChain)
         {
-            string name = _protocolBodyParameter == null
+            string name = !_orderedParameters.Where(parameter => parameter.Convenience != KnownParameters.CancellationTokenParameter).ToList().Exists(parameter => !IsParameterTypeSame(parameter.Convenience, parameter.Protocol))
```

Comment:
since we are using linq, better to use `Any` instead of `ToList().Exists(predicate)`, they should have the same result.
And as a complicated condition like this, please create a method to make this more readable
```suggestion
            string name = !_orderedParameters.Where(parameter => parameter.Convenience != KnownParameters.CancellationTokenParameter).Any(parameter => !IsParameterTypeSame(parameter.Convenience, parameter.Protocol))
```
And BTW, semantically, `!array.Any(predicate)` is equivalent to `array.All(predicate)`. Avoiding to use double negative could make our code cleaner